src/pybind/mgr/dashboard/frontend/src/app/shared/api/grafana.service.ts

```diff
@@ -0,0 +1,26 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { ApiModule } from './api.module';
+
+@Injectable({
+  providedIn: ApiModule
+})
+export class GrafanaService {
+  port: number;
+  protocol: string;
+  host: string;
+  constructor(private http: HttpClient) {}
+
+  getProtocol() {
```

Comment:
Since all 3 methods are always called together I would suggest calling them all at the same time.
If you think it would be better to keep the 3 methods, you can also implement the change in the grafana component.
Here is an example:
https://medium.com/@swarnakishore/performing-multiple-http-requests-in-angular-4-5-with-forkjoin-74f3ac166d61

Reply:
ack